src/os/ui/layer/compare/layercompareui.js

```diff
@@ -462,6 +469,12 @@ const windowId = 'compare-layers';
  * @param {!LayerCompareOptions} options The layer compare options.
  */
 const launchLayerCompare = (options) => {
+  const featureCount = countFeatures(options.left) + countFeatures(options.right);
+  if (featureCount > getMaxFeatures('2d')) {
+    AlertManager.getInstance().sendAlert('Using Layer Compare with the current data volume may degrade performance ' +
```

Comment:
An alert is informative for the user, but doesn't prevent the bad behavior from happening. If a user opens compare with 150k features on a slow machine, they might crash their browser.

I think it would be better to treat this like our 3D -> 2D switch in the same circumstance where we first pop up a warning confirmation to make sure they want to continue. You can look at `os.webgl.launch2DPerformanceDialog` for how that is currently done. We'll want the dialog's text to be specific to Layer Compare, but otherwise it should be largely the same. If they confirm, launch the dialog. If they cancel, do nothing.